source/torrent/torrent_service.py

```diff
@@ -0,0 +1,192 @@
+import queue
+import threading
+import requests
+import hashlib
+import bencode
+import urllib.parse
+
+from utils.logger import setup_logger
+from typing import List
+
+
+from torrent.torrent_item import TorrentItem
+from jackett.jackett_result import JackettResult
+
+class TorrentService:
+    def __init__(self):
+        self.logger = setup_logger(__name__)
+        self.__session = requests.Session()
+
+    def convert_and_process(self, results: List[JackettResult]):
+        threads = []
+        torrent_items_queue = queue.Queue() 
+
+        def thread_target(result: JackettResult):
+            torrent_item = result.convert_to_torrent_item()
+
+            if torrent_item.link.startswith("magnet:"):
+                processed_torrent_item = self.__process_magnet(torrent_item)
+            else:
+                processed_torrent_item = self.__process_web_url(torrent_item)
+
+            torrent_items_queue.put(processed_torrent_item)
+        
+        for result in results:
+            threads.append(threading.Thread(target=thread_target, args=(result,)))
+
+        for thread in threads:
+            thread.start()
+
+        for thread in threads:
+            thread.join()
+
+        torrent_items_result = []
+
+        while not torrent_items_queue.empty():
+            torrent_items_result.append(torrent_items_queue.get())
+
+        return torrent_items_result
+
+    def __process_web_url(self, result: TorrentItem):
+        response = self.__session.get(result.link, allow_redirects=False)
+
+        if response.status_code == 200:
+            return self.__process_torrent(result, response.content)
+        elif response.status_code == 302:
+            return self.__process_magnet(response.headers['Location'])
```

Comment:
process_magnet needs a torrent item, not a string

Reply:
Great catch!